ihm/multi_state_scheme.py

```diff
@@ -0,0 +1,225 @@
+    could be resolved.
+
+    :param begin_state: The start state of the connectivity.
+    :type begin_state: :class:`ihm.model.State`
+    :param end_state: The end state of the connectivity. Can be None in case
+     of states that are not connected to others.
+    :type end_state: :class:`ìhm.model.State`
+    :param details: Details to the connectivity.
+    :param dataset_group: The DatasetGroup that was used to obtain information
+    on the connectivity.
+    :type dataset_group: :class:`ìhm.dataset.DatasetGroup`
+    :param kinetic_rate: A kinetic rate assigned to the connectivity.
+    :type kinetic_rate: :class:`KineticRate`
+    :param relaxation_time: A relaxation time assigned to the connectivity.
+    :type relaxation_time: :class:`RelaxationTime`
+    """
+    def __init__(self, begin_state, end_state=None, details=None,
+                 dataset_group=None, kinetic_rate=None, relaxation_time=None):
+        self.begin_state = begin_state
+        self.end_state = end_state
+        self.details = details
+        self.dataset_group = dataset_group
+        self.kinetic_rate = kinetic_rate
+        self.relaxation_time = relaxation_time
+        # The _assigned_to_scheme variable tracks whether the connectivity
+        # has been assigned to a scheme. This is to ensure that each
+        # connectivity is only assigned to a single scheme.
+        self._assigned_to_scheme = False
+
+    def set_assigned_to_scheme(self):
+        self._assigned_to_scheme = True
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+
+class KineticRate(object):
+    """A kinetic rate that can be assigned to a connectivity.
+    The kinetic rate could be either a transition_rate_constant or
+    an equilibrium_constant.
+    In case of an equilibrium_constant, the equilibrium_constant_unit and
+    equilibrium_constant_determination_method are required.
+
+    :param float transition_rate_constant: A transition rate constant
+      describing the exchange between two states. Unit: per second.
+    :param float equilibrium_constant: An equilibrium constant describing the
+      exchange between two states
+    :param str equilibrium_constant_determination_method:
+      The method how the equilibrium_constant was determined.
+      Options are: ['equilibrium constant is determined from population',
+      'equilibrium constant is determined from kinetic rates, kAB/kBA',
+      'equilibrium constant is determined from another method not listed']
+    :param str equilibrium_constant_unit: Unit of the equilibrium constant.
+      Depending on what the process described,a unit might be applicable or not
+    :param str details: Details on the kinetic rate.
+    :param dataset_group: The DatasetGroup used to determine the kinetic rate.
+    :type dataset_group: :class:`ihm.dataset.DatasetGroup`
+    :param external_file: External file containing measurement data
+      for the kinetic rate.
```

Comment:
No type is declared here for `external_file`. And convention in ihm for this parameter is to call it `name`. See, e.g. the `ihm.model.Ensemble` class.